community/kernel/src/main/java/org/neo4j/kernel/impl/store/format/RecordFormats.java

```diff
@@ -44,9 +48,15 @@ public Factory( String key, String... altKeys )
 
     String storeVersion();
 
+    /**
+     * @return format generation, with the intent of usage being that a store can migrate to a newer or
+     * same generation, but not to an older generation.
+     */
+    int generation();
```

Comment:
to include those guys we will need more complex schema with base version (neo version) and subversion for formats at lest - that sound complex, agree, so just please add how those numbers for generations need to be chosen for future formats into javadocs and lets leave it as it is